api/src/main/java/io/minio/messages/LifecycleRule.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio.messages;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.simpleframework.xml.Element;
+import org.simpleframework.xml.Root;
+
+/** Helper class to denote Rule information for {@link LifecycleConfiguration}. */
+@Root(name = "Rule")
+public class LifecycleRule {
+  @Element(name = "AbortIncompleteMultipartUpload", required = false)
+  private AbortIncompleteMultipartUpload abortIncompleteMultipartUpload;
+
+  @Element(name = "Expiration", required = false)
+  private Expiration expiration;
+
+  @Element(name = "Filter", required = false)
+  private RuleFilter filter;
+
+  @Element(name = "ID", required = false)
+  private String id;
+
+  @Element(name = "NoncurrentVersionExpiration", required = false)
+  private NoncurrentVersionExpiration noncurrentVersionExpiration;
+
+  @Element(name = "NoncurrentVersionTransition", required = false)
+  private NoncurrentVersionTransition noncurrentVersionTransition;
+
+  @Element(name = "Status")
+  private Status status;
+
+  @Element(name = "Transition", required = false)
+  private Transition transition;
+
+  /** Constructs new server-side encryption configuration rule. */
+  public LifecycleRule(
```

Comment:
Instead of a constructor like this, it might be better to introduce a builder-like mechanism to build the rule. e.g. instead of 

```
new LifecycleRule(
  Status.ENABLED,
  null,
  new Expiration((ZonedDateTime) null, 365, null),
  new RuleFilter("logs/"),
  "rule2",
  null,
  null,
  null));
```

Something like this:

```
LifecycleRule.builder().
  id("rule2").
  status(Status.ENABLED).
  ruleFilter("logs/").
  expiration(new Expiration((ZonedDateTime) null, 365, null)).
  build()
```

Reply:
Builder is surely helpful for users. However this constructor still serves to prevent invalid values in XML response from server (We see many weirdly implemented S3 servers these days).

The builder is an improvement for many such messages. I think it can be added on user requests.